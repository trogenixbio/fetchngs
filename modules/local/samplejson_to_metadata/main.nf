process SAMPLEJSON_TO_METADATA {
    tag "$sample_json"
    label 'error_retry'

    conda "conda-forge::python=3.9.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    path sample_json
    path metadata_schema
    path mappings_json

    output:
    path "metadata.json"        , emit: metadata_json
    path "*.tsv"                , emit: metadata_tsv
    path "versions.yml"         , emit: versions

    script:
    """
    samplejson_to_metadata.py \\
        ${sample_json} \\
        ${metadata_schema} \\
        metadata.json \\
        ${mappings_json}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """
}
