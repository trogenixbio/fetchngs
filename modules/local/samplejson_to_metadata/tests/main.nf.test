nextflow_process {

    name "Test Process SAMPLEJSON_TO_METADATA"
    script "../main.nf"
    process "SAMPLEJSON_TO_METADATA"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "samplesheet.json", checkIfExists: true)
                input[1] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "ERC000011-TRGX01.json", checkIfExists: true)
                input[2] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "mappings.json", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
