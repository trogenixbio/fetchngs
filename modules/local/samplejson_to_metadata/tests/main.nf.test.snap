{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "metadata.json:md5,adcead9f8080ea6600b25c14cc26e7ed"
                ],
                "1": [
                    [
                        "experiment.tsv:md5,ffc895156d0c8d3e8c7a77f9ccb777f1",
                        "run.tsv:md5,c9e0cb88280cc9eb12ccd4c6096d49d0",
                        "sample.tsv:md5,21fdfa6576acbf87e33d4f338259e063",
                        "study.tsv:md5,5b09d8d0d4896435c8653d7aa381d18e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,3b7bcd2de39fd876cc2e6e4e389dbd9c"
                ],
                "metadata_json": [
                    "metadata.json:md5,adcead9f8080ea6600b25c14cc26e7ed"
                ],
                "metadata_tsv": [
                    [
                        "experiment.tsv:md5,ffc895156d0c8d3e8c7a77f9ccb777f1",
                        "run.tsv:md5,c9e0cb88280cc9eb12ccd4c6096d49d0",
                        "sample.tsv:md5,21fdfa6576acbf87e33d4f338259e063",
                        "study.tsv:md5,5b09d8d0d4896435c8653d7aa381d18e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3b7bcd2de39fd876cc2e6e4e389dbd9c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-21T16:33:35.285428223"
    }
}